### DISCUSSION TITLE
Policy issue on new accounts

### DISCUSSION THREAD

**Discussion by ashiva** [supabase/supabase#2095] [Questions] [answered]
Running into an issue using Supabase Auth + Policies, I have an existing account that I can create new records/rows from fine, but when trying to signup with a new account and create a new record that only that user should be able to create i run into this error:

DB Error: insert or update on table "gigs" violates foreign key constraint "gigs_user_id_fkey"

![image](https://user-images.githubusercontent.com/4908663/123493217-a7cbd780-d5d0-11eb-8003-2fedbf7e0f06.png)

**Comment by silentworks** [comment, marked as answer]
Do you have a foreign key constraint on the gigs table? This looks more like an issue with the table having a constraint that doesn't allow the same user_id inserting/updating this table, check to be sure that its trying to insert and not update that record. If possible can you share your database schema for the `gigs` table here.

**Reply by w3b6x9** [reply]
@ashiva I'm the one who replied to your support ticket about this but wanted to briefly mention it here as well.

Your RLS is fine. Like @silentworks mentioned, check your constraints for `user_id` b/c that's where the issue is.

**Comment by ashiva** [comment]
Thanks,

I’ve tested both locally and on production - and I think I’ve narrowed down the issue to new accounts only - I get that error when trying to create a record with a newly registered account after confirming the signup link then immediately trying to create the record I get this error. However, if I wait a while then try again it works on the same account - could it be some server delay at the database level?

**Reply by silentworks** [reply]
Can you share some code as to how you are trying to create a new record immediately after confirming the sign up link please?

**Reply by ashiva** [reply]
Figured it out, it was due to needing to initialize a profile record on first sign in since creating a `gig` pulled from profile table.

Same solution as https://github.com/supabase/supabase/discussions/1751
